proposals/lip-0050.md

```diff
@@ -225,10 +302,32 @@ reclaimParamsSchema = {
 ```python
 def verify(trs: Transaction) -> None:
     legacyAddress = getLegacyAddress(trs.senderPublicKey)
-    if legacyAccounts(legacyAddress) does not exists :
+    if legacyAccounts(legacyAddress) is empty:
+        address = SHA-256(trs.senderPublicKey)[:NUM_BYTES_ADDRESS]
+        emitPersistentEvent(
+            module = MODULE_NAME_LEGACY,
+            name = EVENT_NAME_RECLAIM_FAILED,
+            data = {
+                "legacyAddress": legacyAddress,
+                "address": address,
+                "reason": RECLAIM_FAILED_NO_LEGACY_ACCOUNT
+            },
+            topics = [legacyAddress, address]
+        )
         raise Exception('Public key does not correspond to a reclaimable account.')
 
     if legacyAccounts(legacyAddress).balance != trs.params.amount:
+        address = SHA-256(trs.senderPublicKey)[:NUM_BYTES_ADDRESS]
+        emitPersistentEvent(
+            module = MODULE_NAME_LEGACY,
+            name = EVENT_NAME_RECLAIM_FAILED,
+            data = {
+                "legacyAddress": legacyAddress,
+                "address": address,
+                "reason": RECLAIM_FAILED_INVALID_AMOUNT
+            },
+            topics = [legacyAddress, address]
+        )
```

Comment:
same, I don't think we should emit here.

Reply:
Indeed, removed events from verification.